proxy.go

```diff
@@ -55,25 +50,55 @@ func (h *proxyHandler) handleConnect(proxyResp http.ResponseWriter, proxyReq *ht
 		sendError(proxyResp, "Failed to hijack connection", http.StatusInternalServerError, err)
 		return
 	}
-
-	log.Debug("hijacked connection, writing connect request")
-
 	// We're responsible for closing the client connection since we hijacked it. But since we're
 	// just piping data back and forth, we can give that responsibility back to the server, which
 	// will close the connection when the request is done.
 	context.AfterFunc(proxyReq.Context(), func() { clientConn.Close() })
+	log.Debug("hijacked connection")
 
-	// Create a new CONNECT request to send to the proxy server.
-	connectReq, err := backend.NewRequestWithHeaders(
-		proxyReq.Context(),
-		http.MethodConnect,
-		proxyReq.URL.String(),
-		nil,
-	)
-	connectReq.Header.Set(authTokenHeader, h.authToken)
-	if err = connectReq.Write(targetConn); err != nil {
-		sendError(proxyResp, "Failed to write connect request to proxy", http.StatusInternalServerError, err)
-		return
+	var targetConn transport.StreamConn
+	var proxylessFailedOrSkipped bool
+
+	if h.proxylessDialer != nil {
+		targetConn, err = h.proxylessDialer.DialStream(proxyReq.Context(), proxyReq.Host)
```

Comment:
We need a timeout here or we'll never try to dial through the proxy when proxyless dials hang.  I think 10 seconds should be okay.  Sounds like technically a TCP connection could successfully establish even with 21 seconds of latency and retries, but I can't imagine that's common: https://forum.netgate.com/topic/85793/tcp-timeouts.